packages/vsx-registry/src/browser/vsx-extension.tsx

```diff
@@ -0,0 +1,441 @@
+const downloadCompactFormatter = new Intl.NumberFormat(undefined, { notation: 'compact', compactDisplay: 'short' } as any);
+
+export class VSXExtensionComponent extends AbstractVSXExtensionComponent {
+    render(): React.ReactNode {
+        const { iconUrl, publisher, displayName, description, version, downloadCount, averageRating } = this.props.extension;
+        const iconStyle: React.CSSProperties = {};
+        if (iconUrl) {
+            iconStyle.backgroundImage = `url('${iconUrl}')`;
+        }
+        return <div className='theia-vsx-extension'>
+            <div className='theia-vsx-extension-icon' style={iconStyle} />
+            <div className='theia-vsx-extension-content'>
+                <div className='title'>
+                    <div className='noWrapInfo'>
+                        <span className='name'>{displayName}</span> <span className='version'>{version}</span>
+                    </div>
+                    <div className='stat'>
+                        {downloadCount && <span className='download-count'><i className='fa fa-download' />{downloadCompactFormatter.format(downloadCount)}</span>}
+                        {averageRating && <span className='average-rating'><i className='fa fa-star' />{averageRating.toFixed(1)}</span>}
+                    </div>
+                </div>
+                <div className='noWrapInfo theia-vsx-extension-description'>{description}</div>
+                <div className='theia-vsx-extension-action-bar'>
+                    <span className='noWrapInfo theia-vsx-extension-publisher'>{publisher}</span>
+                    {this.renderAction()}
+                </div>
+            </div>
+        </div >;
+    }
+}
+
+export class VSXExtensionEditorComponent extends AbstractVSXExtensionComponent {
+    render(): React.ReactNode {
+        const { builtin, preview, id, iconUrl, publisher, displayName, description, averageRating, downloadCount, repository, license, readme } = this.props.extension;
+        const iconStyle: React.CSSProperties = {};
+        if (iconUrl) {
+            iconStyle.backgroundImage = `url('${iconUrl}')`;
+        }
+        return <React.Fragment>
+            <div className='header'>
+                <div className='icon-container' style={iconStyle} />
+                <div className='details'>
+                    <div className='title'>
+                        <span title='Extension name' className='name' onClick={this.openExtension}>{displayName}</span>
+                        <span title='Extension identifier' className='identifier'>{id}</span>
+                        {preview && <span className='preview'>Preview</span>}
+                        {builtin && <span className='builtin'>Built-in</span>}
+                    </div>
+                    <div className='subtitle'>
+                        <span title='Publisher name' className='publisher' onClick={this.searchPublisher}>{publisher}</span>
```

Comment:
Please align with open-vsx.org and display the `publishedBy` info here instead of the publisher, along with an icon indicating the `namespaceAccess` info (`public` or `restricted`).

https://github.com/eclipse/openvsx/blob/74dd60eebc6855411b91f67ff809c15d9ad68a7a/webui/src/pages/extension-detail/extension-detail.tsx#L200-L242

Reply:
I actually find open VSX UI confusing, publisher should be a filed from package.json, but it shows the person who published and never show `namespace` field at all.

I cannot do it here because I have to care about locally installed extensions as well and for them it is `publisher` field from package.json.

Reply:
I will render an icon before namespace to indicate access (globe for public, shield for restricted), title will explain and on click the actual publisher will be opened. For built-in extension there won’t be such icon.